### PR TITLE
More robust table styleing

### PR DESCRIPTION
**Pull request by mattheu**
Handle `thead`, `tfoot` and `caption`.

### CODE REVIEW

File: src/styles/base/_tables.scss [hunk 1]
```diff
@@ -16,8 +16,13 @@ td {
 	}
 }
 
-tr:last-child td,
-tr:last-child th {
+tbody:last-child > tr:last-child td {
```
Comment (sambulance) [inline src/styles/base/_tables.scss]: I think these styles are _super_ specific, and assume that all table are marked up with a `tbody`/`tfoot` etc. Is there a border collapse issue?

File: src/styles/style-guide.scss [hunk 2]
```diff
@@ -6,6 +6,21 @@
 @import "style";
 @import "components/Swatch";
 
+.container {
```
Comment (sambulance) [inline src/styles/style-guide.scss]: This is all in the other PR https://github.com/humanmade/hm-pattern-library/pull/73 - perhaps we can close

File: src/styles/base/_tables.scss (hunk 1 above)
Comment (mattheu) [inline src/styles/base/_tables.scss]: I suppose I only tested chrome... but if you inspect a table - it actually adds a `tbody` if you haven't specified one

Comment (sambulance) [inline src/styles/base/_tables.scss]: Hmm.. Feel like it has potential to break to me, but if you're happy with it then 👍

Comment (mattheu) [inline src/styles/base/_tables.scss]: Hmm maybe. I can test it some more 😀

Comment (mattheu) [inline src/styles/base/_tables.scss]: Added a test case for with/without tbody/thead/tfoot. I'm confident this isn't going to break things!  Tested in a few (modern) browsers and all looks good.

**Pull request merged**